### PR TITLE
[DEV-1164] buy process

### PR DESCRIPTION
**Pull request by Krysh90**

### CODE REVIEW

File: src/validations.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        return 'validation.pattern_invalid';
+      }
+
+      return true;
+    });
+  }
+
+  public Custom = (validator: (value: any) => true | string) => ({
+    validate: (val: any) => (typeof validator(val) == 'boolean' ? validator(val) : i18n.t(validator(val) as string)),
+  });*/
```
[the remark this answers is not in the document]
Comment (Krysh90) [reply src/validations.ts:72]: got deactivated for now, will be added as soon as everything works and are needed. Currently no form has any working rules. We need work on it today and tomorrow to activate them step by step

File: public/blocked-iban.json [hunk 2]
```diff
@@ -0,0 +1 @@
+[]
```
[the remark this answers is not in the document]
Comment (Krysh90) [reply public/blocked-iban.json:1]: fair enough, don't know why I deleted them

File: src/stories/form/StyledModalDropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  detailLabelFunc?: (item: any) => string;
+  modal: {
+    heading: string;
+    items: any;
+    itemContent: (item: any) => JSX.Element;
+    form?: (onFormSubmit: (item: any) => void) => JSX.Element;
+  };
+}
+
+const StyledModalDropdown = forwardRef<any, any>(
```
[the remark this answers is not in the document]
Comment (Krysh90) [reply src/stories/form/StyledModalDropdown.tsx:18]: did try that, couldn't get it to work properly. But will try again

File: src/stories/form/StyledInput.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import { forwardRef } from 'react';
+import { Controller } from 'react-hook-form';
+import { ControlProps } from './Form';
+
+interface StyledInputProps extends ControlProps {
+  placeholder?: string;
+}
+
+const StyledInput = forwardRef<any, any>(
```
Comment (davidleomay) [inline src/stories/form/StyledInput.tsx:9]: `const StyledInput = forwardRef<HTMLInputElement, StyledInputProps>(`

**Review by davidleomay — APPROVED** [review]

**Pull request merged**